### PR TITLE
Fix bug when JSON is a single value

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2506

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/xmldata.bal [hunk 1]
```diff
@@ -43,6 +43,10 @@ public type JsonOptions record {
 # + return - XML representation of the given JSON if the JSON is
 # successfully converted or else an `xmldata:Error`
 public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error {
+    if (jsonValue is string || jsonValue is boolean || jsonValue is decimal || jsonValue is float ||
```
Comment (MadhukaHarith92) [inline ballerina/xmldata.bal:46]: Shall we move this to https://github.com/ballerina-platform/module-ballerina-xmldata/blob/5b9b7bee4099e8c1b9008c9a94015d1fb79f16a8/ballerina/xmldata.bal#L67 (else block) and change it like

```
if !(jsonValue is null) {
    return xml:createText(jsonValue.toString());
}
```

Comment (kalaiyarasiganeshalingam) [reply ballerina/xmldata.bal]: Can't move to Line 67, so I moved that to [here](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/359/files#diff-ecda4f4acb566335c5b8a6598ba4ffc0002c3cf114c6867b20c16bb26d67780bR70). But, IMO, If JSON has a single value, we can return a text value without [these checks](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/359/commits/89e4aae074d0190cb2410f9d1ea94e5285a0195e#diff-ecda4f4acb566335c5b8a6598ba4ffc0002c3cf114c6867b20c16bb26d67780bR46). So I added in the first.

File: ballerina/xmldata.bal [hunk 2]
```diff
@@ -66,6 +67,9 @@ public isolated function fromJson(json jsonValue, JsonOptions options = {}) retu
             }
         }
     }
+    if !(jsonValue is null) {
```
Comment (daneshk) [inline ballerina/xmldata.bal:70]:
```suggestion
    if jsonValue not is null {
```

Comment (daneshk) [reply ballerina/xmldata.bal:70]: Are we already handling null values?

File: ballerina/tests/xml_from_json_test.bal [hunk 3]
```diff
@@ -130,11 +130,39 @@ isolated function testNodeNameNull() returns error? {
 }
 isolated function testJsonAsInt() {
     json data = 5;
+    xml expected = xml `5`;
     xml|Error? result = fromJson(data);
```
Comment (daneshk) [inline ballerina/tests/xml_from_json_test.bal:134]: shall we `check` and return the error here

File: ballerina/tests/xml_from_json_test.bal [hunk 4]
```diff
@@ -130,11 +130,39 @@ isolated function testNodeNameNull() returns error? {
...
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testJsonAsString() {
+    json data = "data";
+    xml expected = xml `data`;
+    xml|Error? result = fromJson(data);
```
Comment (daneshk) [inline ballerina/tests/xml_from_json_test.bal:148]: shall we check and return the error here

File: ballerina/tests/xml_from_json_test.bal [hunk 5]
```diff
@@ -130,11 +130,39 @@ isolated function testNodeNameNull() returns error? {
...
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testJsonAsBoolean() {
+    json data = false;
+    xml expected = xml `false`;
+    xml|Error? result = fromJson(data);
```
Comment (daneshk) [inline ballerina/tests/xml_from_json_test.bal:162]: shall we `check` and return the error here

File: ballerina/xmldata.bal (hunk 2 above)
Comment (MadhukaHarith92) [reply ballerina/xmldata.bal:70]: Currently, an error (`"failed to parse xml"`) is being returned for `null` values. I think this should be changed to return an empty sequence. WDYT.

Comment (kalaiyarasiganeshalingam) [reply ballerina/xmldata.bal]: Fixed in 1da75b6

File: ballerina/tests/xml_from_json_test.bal [hunk 6]
```diff
@@ -128,28 +128,53 @@ isolated function testNodeNameNull() returns error? {
...
+    if result is xml {
+        test:assertEquals(result, expected, msg = "testJsonAsString result incorrect");
+    } else {
+        test:assertFail("Result is not mismatch");
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
```
Comment (MadhukaHarith92) [inline ballerina/tests/xml_from_json_test.bal:172]: Shall we add a test case with a `decimal` value?

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

**Review by daneshk — APPROVED** [review]

**Pull request merged**